nlp_scripts/model_monitoring/model_performance.py

```diff
@@ -1,361 +1,425 @@
+                        pl.lit(
+                            project_grp_df[f"completely_matched_{category}"].mean()
+                        ).alias(f"completely_matched_{category}_mean")
+                    )
+                if f"missing_{category}" in project_grp_df.columns:
+                    temp_df = temp_df.with_columns(
+                        pl.lit(project_grp_df[f"missing_{category}"].mean()).alias(
+                            f"missing_{category}_mean"
+                        )
+                    )
+                if f"wrong_{category}" in project_grp_df.columns:
+                    temp_df = temp_df.with_columns(
+                        pl.lit(project_grp_df[f"wrong_{category}"].mean()).alias(
+                            f"wrong_{category}_mean"
+                        )
+                    )
+
+            temp_df = temp_df.with_columns(
+                pl.lit(project_id).alias("project_id"),
+                pl.lit(datetime.date.today()).alias("generated_at"),
+            )
+            final_df = pl.concat([final_df, temp_df], how="vertical")
         return final_df
 
 
 if __name__ == "__main__":
-    df = (
-        pd.read_csv("csvfiles/sampled_data_with_predictions_testset.csv")
-        .sample(n=50, random_state=1234)
-        .reset_index()
+    # example dataframe
+    pl_df = pl.DataFrame(
+        {
+            "project_id": [1111, 1234],
+            "sectors": [
+                json.dumps(
+                    [
+                        "first_level_tags->sectors->Agriculture",
+                        "first_level_tags->sectors->Shelter",
+                    ]
+                ),
+                json.dumps(["first_level_tags->sectors->Agriculture"]),
+            ],
+            "pillars_1d": [
+                json.dumps(["first_level_tags->pillars_1d->Casualties"]),
+                json.dumps(["first_level_tags->pillars_1d->Context"]),
+            ],
+            "sectors_pred": [
+                json.dumps(
+                    [
+                        "first_level_tags->sectors->Agriculture",
+                        "first_level_tags->sectors->Protection",
+                    ]
+                ),
+                json.dumps(["first_level_tags->sectors->Agriculture"]),
+            ],
+            "pillars_1d_pred": [
+                json.dumps(["first_level_tags->pillars_1d->Casualties"]),
+                json.dumps(["first_level_tags->pillars_1d->Context"]),
+            ],
+        }
     )
-    modelperf = ModelPerformance(df)
-
-    df1 = modelperf.project_wise_perf_metrics()
-    # print(df1)
-    df2 = modelperf.per_tag_perf_metrics()
-    # print(df2)
-    df3 = modelperf.all_projects_perf_metrics()
-    # print(df3)
-    df4 = modelperf.calculate_ratios()
-    # print(df4)
-    df5 = modelperf.per_project_calc_ratios()
-    # print(df5)
+    modelperf = ModelPerformance(dataframe=pl_df)
+    modelperf.label_transform()
+    print(modelperf.projectwise_perf_metrics())
```

Comment:
```suggestion
```

Reply:
It's inside __main__ so should not matter.